appengine-jetty-managed-runtime/src/main/java/com/google/apphosting/vmruntime/jetty9/VmRuntimeWebAppContext.java

```diff
@@ -262,129 +260,96 @@ public boolean isTrustedRemoteAddr(String remoteAddr) {
-    CommitDelayingResponse wrappedResponse;
-    if (httpServletResponse instanceof CommitDelayingResponse) {
-      wrappedResponse = (CommitDelayingResponse) httpServletResponse;
-    } else {
-      wrappedResponse = new CommitDelayingResponse(httpServletResponse);
-    }
-
-    try {
-      ApiProxy.setEnvironmentForCurrentThread(requestSpecificEnvironment);
-      // Check for SkipAdminCheck and set attributes accordingly.
-      VmRuntimeUtils.handleSkipAdminCheck(request);
-      // Change scheme to HTTPS based on headers set by the appserver.
-      setSchemeAndPort(baseRequest);
-      // Forward the request to the rest of the handlers.
-      super.doScope(target, baseRequest, httpServletRequest, wrappedResponse);
-    } finally {
-      try {
-        // Interrupt any remaining request threads and wait for them to complete.
-        VmRuntimeUtils.interruptRequestThreads(
-            requestSpecificEnvironment, VmRuntimeUtils.MAX_REQUEST_THREAD_INTERRUPT_WAIT_TIME_MS);
-        // Wait for any pending async API requests to complete.
-        if (!VmRuntimeUtils.waitForAsyncApiCalls(requestSpecificEnvironment,
-            new HttpServletResponseAdapter(wrappedResponse))) {
-          logger.warning("Timed out or interrupted while waiting for async API calls to complete.");
-        }
-        if (!response.isCommitted()) {
-          // Flush and set the flush count header so the appserver knows when all logs are in.
-          VmRuntimeUtils.flushLogsAndAddHeader(response, requestSpecificEnvironment);
-        } else {
-          throw new ServletException("Response for request to '" + target
-              + "' was already commited (code=" + httpServletResponse.getStatus()
-              + "). This might result in lost log messages.'");
-        }
-      } finally {
-        try {
-          // Complete any pending actions.
-          wrappedResponse.commit();
-        } finally {
-          // Restore the default environment.
-          ApiProxy.setEnvironmentForCurrentThread(defaultEnvironment);
-        }
-      }
+  public RequestContext getRequestContext(Request baseRequest) {
+    RequestContext requestContext = (RequestContext)baseRequest.getAttribute(RequestContext.class.getName());
+    if (requestContext==null) {
+      // No instance found, so create a new environment
+      requestContext=new RequestContext(baseRequest);
+      baseRequest.setAttribute(RequestContext.class.getName(), requestContext);
     }
+    return requestContext;
   }
 
-  // N.B.(schwardo): Yuck. Jetty hardcodes all of this logic into an
-  // inner class of ContextHandler. We need to subclass WebAppContext
-  // (which extends ContextHandler) and then subclass the SContext
-  // inner class to modify its behavior.
   /**
    * ServletContext for VmRuntime applications.
+   * TODO is this still needed? If no securityManager super.getClassLoader() is equivalent
    */
   public class VmRuntimeServletContext extends Context {
```

Comment:
@ludoch The only reason we now extend Context is to replace the call to getClassloader.  If no securityManager is present, then this is equivalent.  What is the securityManager environment and do we still need this?